lib/client/conntest/database/database.go

```diff
@@ -28,16 +28,12 @@ type PingParams struct {
 	Port int
 	// Username is the user to be used to login into the database.
 	Username string
-	// Database is the database name to be used to login into the database.
-	Database string
+	// DatabaseName is an optional database name to be used to login into the database.
+	DatabaseName string
 }
 
 // CheckAndSetDefaults validates and set the default values for the Ping.
 func (req *PingParams) CheckAndSetDefaults() error {
-	if req.Database == "" {
-		return trace.BadParameter("missing required parameter Database")
-	}
-
```

Comment:
I was tempted to remove this check altogether, since we already get required db names/users while creating the pinger in `newPing`. But in MySQL database name is not required.

Reply:
Can we pass the DatabaseProtocol field and have a different validation for MySQL and Postgres ?

Reply:
I think we should always require the Database Name because we are configuring a User's DB Name traits.
If they enter an invalid one, they must be warned about it.

My understanding is that we are already validating it in the PingParams creation, so there's no need to validate it again here.

But then, I see no reason to only remove the Database. I think we should either validate both DatabaseName and User or validate none.

Reply:
Resolving as discussed in the other comment
https://github.com/gravitational/teleport/pull/20177#discussion_r1069155149

Reply:
I changed it to check by protocol. I wanted to delete the checkandset func altogether, but since `PingParams` is an exported type that was not a good idea (just to be safe in the future if other callers use it).